### PR TITLE
Update Text support Theming

### PR DESCRIPTION
**Pull request by williamkolmacka**
Added support for theming
Rewrited to styled components
Added flow strict type
Added constants for props

### CODE REVIEW

File: src/Text/__tests__/index.test.js [hunk 1]
```diff
@@ -2,15 +2,34 @@
...
+          {children}
+        </Text>
+      </ThemeProvider>,
+    );
+    expect(
+      component
+        .children()
+        .children()
+        .exists(),
+    ).toBe(true);
```
Comment (jaroslav-kubicek) [inline src/Text/__tests__/index.test.js]: Pls check that it contains `Children text`

File: src/Text/__tests__/__snapshots__/index.test.js.snap [hunk 2]
```diff
@@ -1,27 +1,375 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`Rendered with children Should contain text 1`] = `
-<React.Fragment>
-  <p
-    className="jsx-1052758011 "
+exports[`Text should match snapshot 1`] = `
+<ThemeProvider
```
Comment (jaroslav-kubicek) [inline src/Text/__tests__/__snapshots__/index.test.js.snap]: this snapshot doesn't make much sense, please keep the old one - it actually shouldn't be affected by changes in this PR (so it the test, do something like `expect(wrapper.find('Text')).toMatchSnapshot()`)

File: src/Text/__tests__/index.test.js (hunk 1 above)
Comment (williamkolmacka) [reply src/Text/__tests__/index.test.js]: Fixed.

File: src/Text/__tests__/__snapshots__/index.test.js.snap (hunk 2 above)
Comment (williamkolmacka) [reply src/Text/__tests__/__snapshots__/index.test.js.snap]: Fixed.

File: src/Text/index.js [hunk 3]
```diff
@@ -1,79 +1,67 @@
...
-          text-align: ${align};
-          ${uppercase ? `text-transform: uppercase` : ""};
-          ${italic ? `font-style: italic` : ""};
-          margin: 0;
-        `}
-      </style>
-    </scope>,
-  );
+const StyledText = styled(TextElement)`
+  font-family: ${props => props.theme.fontFamily};
```
Comment (wEprous) [inline src/Text/index.js]: Use object destructuring

File: src/Text/index.js [hunk 4]
```diff
@@ -1,79 +1,67 @@
...
+  text-transform: ${props => props.uppercase && `uppercase`};
+  font-style: ${props => props.italic && `italic`};
+  margin: 0;
+`;
 
+const Text = (props: Props) => {
+  const { children, theme } = props;
+  const tokens = {
+    colorText: {
+      primary: theme.colorTextPrimary,
```
Comment (wEprous) [inline src/Text/index.js]: Define constants for type, size and weight

File: src/Text/index.js [hunk 5]
```diff
@@ -1,79 +1,67 @@
...
+import { withTheme } from "theming";
+import createComponentFromTagProp from "react-create-component-from-tag-prop";
 
 type Props = {
-  type: $Keys<typeof colorText>,
-  size: $Keys<typeof sizeText>,
-  weight: $Keys<typeof weightText>,
+  type: "primary" | "secondary" | "attention",
+  size: "small" | "normal" | "large",
+  weight: "regular" | "bold",
```
Comment (wEprous) [inline src/Text/index.js]: Why is here defined `regular` and token from orbit-design-tokens is called `fontWeightNormal` ?

Comment (wEprous) [reply src/Text/index.js]: Fixed.

File: src/Text/index.js (hunk 4 above)
Comment (wEprous) [reply src/Text/index.js]: Fixed.

File: src/Text/index.js (hunk 3 above)
Comment (wEprous) [reply src/Text/index.js]: Fixed.

**Pull request merged**